lib/mix/lib/mix/tasks/help.ex

```diff
@@ -113,11 +113,12 @@ defmodule Mix.Tasks.Help do
   def run(["app:" <> app]) do
     loadpaths!()
     app = String.to_atom(app)
+    Application.ensure_loaded(app)
```

Comment:
Currently we display:

```
$ mix help app:bad
Application bad does not exist or is not loaded
```

if we change it to `Mix.ensure_application!(app)`, we'd see:

```
$ mix help app:bad
** (Mix) The application "bad" could not be found. This may happen if your Operating System broke Erlang into multiple packages and may be fixed by installing the missing "erlang-dev" and "erlang-bad" packages
```

I think former is fine but can see some rationale for latter. And then we could remove tiny bit of error handling code too while at it. Any preference?

Reply:
Let's go with ensure_application!, yeah, I think it will be more fool proof. Perhaps this will make it fail: `mix do compile + help app:wx` but I don't see why anyone would run that anyway...